javascripts/tagmanager.js

```diff
@@ -1635,6 +1635,19 @@
                 throwError: throwError,
                 Container: Container,
                 addContainer: function (containerConfig, templates) {
+                    var mtmSetDebugFlag = urlHelper.getQueryParameter('mtmSetDebugFlag');
+                    if (mtmSetDebugFlag) {
+                        var idSite = encodeURIComponent(containerConfig.idsite);
+                        var containerID = encodeURIComponent(containerConfig.id);
+                        if (mtmSetDebugFlag == 1) {
+                            var date = new Date();
+                            date.setTime(date.getTime() + (7 * 24 * 60 * 60 * 1000));
+                            document.cookie = 'mtmPreviewMode=mtmPreview' + idSite + '_' + containerID + '%3D1;expires=' + date.toUTCString() + ';SameSite=Lax';
```

Comment:
@AltamashShaikh can we urlencode these idsite and container parameters here? also in the `else` below?

Reply:
Do you mean this https://github.com/matomo-org/tag-manager/blob/920fb90be784ba7ec289f2b66ade832d54501752/javascripts/tagmanager.js#L1640 ?

Reply:
👍